### PR TITLE
#CP-26 Setup Redux

### PR DESCRIPTION
**Pull request by samyshyaka**
#### What the PR does
Configure the app to use Redux as a state management library

#### Description of Tasks to be completed

- Set up Redux 
- Create a simple number counter on the landing page using redux as the state management library

#### How this should be manually tested
Clone the project repository linked [here](https://github.com/atlp-rwanda/phantom-fe-codebandits)

```
cd phantom-fe-codebandits/
git checkout ch-setup-redux-cp-26
git fetch origin
npm install
npm run start
run http://localhost:3000 in your browser
```

#### Relevant Trello Card
[#CP-26](https://trello.com/c/94jYzQr3/26-setup-redux)

### CODE REVIEW

**Review by Abdulkeza — DISMISSED** [review]
Hey! on my side app is working well, and Redux too
 you did a great job.

**Review by Yvad60 — DISMISSED** [review]
everything including the counter is working as expected!

File: src/index.js [hunk 1]
```diff
@@ -5,13 +5,25 @@ import {
     Route 
 } from 'react-router-dom';
 import ReactDOM from 'react-dom';
+import { configureStore } from '@reduxjs/toolkit';
+import { Provider } from 'react-redux';
+
+import { counterReducer } from './redux/reducers/counterReducer.js'
 import App from './App.js'
 
+const store = configureStore({
```
Comment (Jkadhuwa) [inline src/index.js]: Move the store configuration into its own file inside the redux folder

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
+import {
+  decrement,
+  increment,
+  reset,
+} from './redux/reducers/counterReducer.js'
 
 const App = ()=> {
+
+    const counter = useSelector((state) => state?.counter);
+    console.log(counter)
```
Comment (Jkadhuwa) [inline src/App.js]: You can remove the log

File: src/App.js [hunk 3]
```diff
@@ -1,11 +1,28 @@
...
+import { useSelector, useDispatch } from "react-redux";
+import {
+  decrement,
+  increment,
+  reset,
+} from './redux/reducers/counterReducer.js'
 
 const App = ()=> {
+
+    const counter = useSelector((state) => state?.counter);
```
Comment (Jkadhuwa) [inline src/App.js]: You can destructure value from state.counter directly (es6)

**Review by Jkadhuwa — CHANGES_REQUESTED** [review]
Nice work so far, attend to the feedback left below.

**Review by Jkadhuwa — APPROVED** [review]

**Pull request merged**